### PR TITLE
libwebrtc: replace MS_ASSERT with MS_ERROR

### LINKED ISSUE #986: [worker] Must replace MS_ASSERT() with a non aborting warn/error log
So it happens that we replaced all those `RTC_DCHECK()` in libwebrtc with `MS_ASSERT()`. However, when in Release mode, `RTC_DCHECK()` in libwebrtc **doesn't not abort the process** (see discussion in https://github.com/versatica/mediasoup/issues/357#issuecomment-1400886101).

In mediasoup `v3` branch we have all these `MS_ASSERT()` in libwebrtc dep:

```
$ ack MS_ASSERT

libwebrtc/system_wrappers/source/field_trial.cc
113:    MS_ASSERT(

libwebrtc/call/rtp_transport_controller_send.cc
75:  MS_ASSERT(bitrate_config.start_bitrate_bps > 0, "start bitrate must be > 0");
90:  MS_ASSERT(observer_ != nullptr, "no observer");
139:    MS_ASSERT(observer_ == nullptr, "observer already set");
265:  MS_ASSERT(!controller_, "controller already set");
266:  MS_ASSERT(!control_handler_, "controller handler already set");
283:  MS_ASSERT(controller_, "controller not set");

libwebrtc/modules/pacing/bitrate_prober.cc
126:  MS_ASSERT(probing_state_ != ProbingState::kDisabled, "probing disabled");
127:  MS_ASSERT(bitrate_bps > 0, "bitrate must be > 0");
144:  MS_ASSERT(cluster.pace_info.probe_cluster_min_bytes >= 0, "cluster min bytes must be >= 0");
197:  MS_ASSERT(!clusters_.empty(), "clusters is empty");
198:  MS_ASSERT(probing_state_ == ProbingState::kActive, "probing not active");
208:  MS_ASSERT(!clusters_.empty(), "clusters is empty");
217:  MS_ASSERT(probing_state_ == ProbingState::kActive, "probing not active");
218:  MS_ASSERT(bytes > 0, "bytes must be > 0");
224:      MS_ASSERT(cluster->time_started_ms == -1, "cluster started time must not be -1");
253:  MS_ASSERT(cluster.pace_info.send_bitrate_bps > 0, "cluster.pace_info.send_bitrate_bps must be > 0");
254:  MS_ASSERT(cluster.time_started_ms > 0, "cluster.time_started_ms must be > 0");

libwebrtc/modules/pacing/paced_sender.cc
102:  MS_ASSERT(packet_counter_ == 0, "packet counter must be 0");
110:  MS_ASSERT(pacing_rate_bps > 0, "pacing rate must be > 0");
124:  MS_ASSERT(pacing_bitrate_kbps_ > 0, "SetPacingRates() must be called before InsertPacket()");

libwebrtc/modules/remote_bitrate_estimator/overuse_estimator.cc
111:  MS_ASSERT(positive_semi_definite, "positive_semi_definite is not true");

libwebrtc/modules/remote_bitrate_estimator/remote_bitrate_estimator_abs_send_time.cc
239:  MS_ASSERT(send_time_24bits < (1ul << 24), "invalid sendTime24bits value");

libwebrtc/modules/remote_bitrate_estimator/inter_arrival.cc
41:  MS_ASSERT(timestamp_delta != nullptr, "timestamp_delta is null");
42:  MS_ASSERT(arrival_time_delta_ms != nullptr, "arrival_time_delta_ms is null");
43:  MS_ASSERT(packet_size_delta != nullptr, "packet_size_delta is null");
92:      MS_ASSERT(*arrival_time_delta_ms >= 0, "arrival_time_delta_ms is < 0");
159:  MS_ASSERT(

libwebrtc/modules/congestion_controller/goog_cc/probe_bitrate_estimator.cc
67:  MS_ASSERT(cluster_id != PacedPacketInfo::kNotAProbe, "cluster_id == kNotAProbe");
94:  MS_ASSERT(
97:  MS_ASSERT(

libwebrtc/modules/congestion_controller/goog_cc/probe_controller.cc
254:  MS_ASSERT(network_available_, "network not available");
255:  MS_ASSERT(state_ == State::kInit, "state_ must be State::kInit");
256:  MS_ASSERT(start_bitrate_bps_ > 0, "start_bitrate_bps_ must be > 0");

libwebrtc/modules/congestion_controller/goog_cc/delay_based_bwe.cc
255:  MS_ASSERT(ssrcs, "ssrcs must be != null");
256:  MS_ASSERT(bitrate, "bitrate must be != null");
```

So we should replace all these `MS_ASSERT()` with a `MS_WARN_TAB(bwe)` (or similar warn/error log) plus a `return xxxx;`.

**NOTE:** This should also be done in `sarumjanuch:bwe_backport` branch in ongoing PR https://github.com/versatica/mediasoup/pull/922

### PR DESCRIPTION
**Pull request by ibc**
- Fixes #986
- Let's use `MS_ERROR` instead of `MS_WARN_TAG(bwe)` to make these errors as visible as possible.
- Sometimes use more modern libwebrtc code such as the usage of `absl::optional` somewhere in code diff.
- Original `RTC_DCHECK` in libwebrtc does NOT abort so we must only show error logs and, depending on each case, do early return or not. If not, it means that code below keeps running with inconsistent data. I've tried to return earlier as much as possible but it's not always possible. In fact, `RTC_CHECK` **does abort* even in Release mode. However, `RTC_DCHECK` just logs it.

### CODE REVIEW

File: worker/deps/libwebrtc/libwebrtc/modules/pacing/bitrate_prober.cc [hunk 1]
```diff
@@ -191,11 +200,12 @@ int BitrateProber::TimeUntilNextProbe(int64_t now_ms) {
...
 
-PacedPacketInfo BitrateProber::CurrentCluster() const {
+absl::optional<PacedPacketInfo> BitrateProber::CurrentCluster() const {
   // RTC_DCHECK(!clusters_.empty());
   // RTC_DCHECK(probing_state_ == ProbingState::kActive);
-  MS_ASSERT(!clusters_.empty(), "clusters is empty");
-  MS_ASSERT(probing_state_ == ProbingState::kActive, "probing not active");
+  if (clusters_.empty() || probing_state_ != ProbingState::kActive) {
+    return absl::nullopt;
+  }
```
Comment (ibc) [inline worker/deps/libwebrtc/libwebrtc/modules/pacing/bitrate_prober.cc:208]: This is how modern libwebrtc code looks.

File: worker/deps/libwebrtc/libwebrtc/modules/pacing/bitrate_prober.cc [hunk 2]
```diff
@@ -205,7 +215,9 @@ PacedPacketInfo BitrateProber::CurrentCluster() const {
 // feasible.
 size_t BitrateProber::RecommendedMinProbeSize() const {
   // RTC_DCHECK(!clusters_.empty());
-  MS_ASSERT(!clusters_.empty(), "clusters is empty");
+  if (clusters_.empty()) {
+    return 0;
```
Comment (ibc) [inline worker/deps/libwebrtc/libwebrtc/modules/pacing/bitrate_prober.cc:219]: Is it ok to return 0 if this happens?

File: worker/deps/libwebrtc/libwebrtc/modules/pacing/bitrate_prober.h [hunk 3]
```diff
@@ -66,7 +66,7 @@ class BitrateProber {
   int TimeUntilNextProbe(int64_t now_ms);
 
   // Information about the current probing cluster.
-  PacedPacketInfo CurrentCluster() const;
+  absl::optional<PacedPacketInfo> CurrentCluster() const;
```
Comment (ibc) [inline worker/deps/libwebrtc/libwebrtc/modules/pacing/bitrate_prober.h:69]: Copied from modern libwebrtc version.

File: worker/deps/libwebrtc/libwebrtc/modules/pacing/paced_sender.cc [hunk 4]
```diff
@@ -185,7 +194,7 @@ void PacedSender::Process() {
   PacedPacketInfo pacing_info;
   absl::optional<size_t> recommended_probe_size;
 
-  pacing_info = prober_.CurrentCluster();
+  pacing_info = prober_.CurrentCluster().value_or(PacedPacketInfo());
```
Comment (ibc) [inline worker/deps/libwebrtc/libwebrtc/modules/pacing/paced_sender.cc:197]: Hope this is ok. Used in libwebrtc modern version somewhere.

File: worker/deps/libwebrtc/libwebrtc/modules/remote_bitrate_estimator/overuse_estimator.cc [hunk 5]
```diff
@@ -108,11 +108,8 @@ void OveruseEstimator::Update(int64_t t_delta,
       E_[0][0] + E_[1][1] >= 0 &&
       E_[0][0] * E_[1][1] - E_[0][1] * E_[1][0] >= 0 && E_[0][0] >= 0;
 
-  MS_ASSERT(positive_semi_definite, "positive_semi_definite is not true");
-
   if (!positive_semi_definite) {
-    MS_ERROR("The over-use estimator's covariance matrix is no longer "
-           "semi-definite.");
+    MS_ERROR("the over-use estimator's covariance matrix is no longer semi-definite");
```
Comment (jmillan) [inline worker/deps/libwebrtc/libwebrtc/modules/remote_bitrate_estimator/overuse_estimator.cc:112]: return missing here?

Comment (ibc) [reply worker/deps/libwebrtc/libwebrtc/modules/remote_bitrate_estimator/overuse_estimator.cc:112]: There is no such a return in libwebrtc.

File: worker/deps/libwebrtc/libwebrtc/modules/pacing/bitrate_prober.cc (hunk 2 above)
Comment (ibc) [reply worker/deps/libwebrtc/libwebrtc/modules/pacing/bitrate_prober.cc:219]: @sarumjanuch thoughts?

Comment (ibc) [reply worker/deps/libwebrtc/libwebrtc/modules/pacing/bitrate_prober.cc:219]: This is "recommended **min** probe size" so we are good.

File: worker/deps/libwebrtc/libwebrtc/modules/pacing/bitrate_prober.cc [hunk 6]
```diff
@@ -242,16 +263,22 @@ void BitrateProber::ProbeSent(int64_t now_ms, size_t bytes) {
...
-  MS_ASSERT(cluster.pace_info.send_bitrate_bps > 0, "cluster.pace_info.send_bitrate_bps must be > 0");
-  MS_ASSERT(cluster.time_started_ms > 0, "cluster.time_started_ms must be > 0");
+  if (cluster.pace_info.send_bitrate_bps <= 0) {
+    MS_ERROR("cluster.pace_info.send_bitrate_bps must be > 0");
+    return 0;
+  }
+  if (cluster.time_started_ms <= 0) {
+    MS_ERROR("cluster.time_started_ms must be > 0");
+    return 0;
+  }
```
Comment (ibc) [inline worker/deps/libwebrtc/libwebrtc/modules/pacing/bitrate_prober.cc]: @jmillan I'm gonna revert these changes because those were `RTC_CHECK` so assertion is needed.

**Review by jmillan — APPROVED** [review]
